### DISCUSSION TITLE
How to access to inside Three.js scene in Model-Viewer

### DISCUSSION THREAD

**Discussion by digityze2023** [google/model-viewer#4553] [Q&A] [answered]
Dear all,
I have a problem using: mvElement[Object.getOwnPropertySymbols(mvElement).find(e => e.description === 'scene')]; 

From the FAQ of https://modelviewer.dev/docs/faq.html#entrydocs-general-questions-api I've heard about it.

Imagine that my <body> is something similar to:

<body>
    <model-viewer
        src="path-to-your-3d-model.glb"
        alt="A 3D Model"
        auto-rotate
        camera-controls
    ></model-viewer>
    
    <script>
        const mvElement = document.querySelector('model-viewer');
     
        const threeJsScene = mvElement[Object.getOwnPropertySymbols(mvElement).find(e => e.description === 'scene')];
        
        const ambientLight = new THREE.AmbientLight(0x404040);
        threeJsScene.add(ambientLight);

    </script>
</body>

The console returns to me "THREE" is not DEFINED..

Could you please help me?

My intentions are: use the threejs library and scene to customize my model (3D configurator) (using the power of threejs and easily display / AR mode of Model-Viewer.js).

**Comment by elalish** [comment, marked as answer]
You'll probably want our separated library: `model-viewer-module.min.js` and import map three.js in if you want to use more of it than we include. [Example](https://modelviewer.dev/examples/postprocessing/#setup).

**Reply by digityze2023** [reply]
Yes, I'd like to use model-viewer and include some Three.js "opportunities". A power up from a point of view.
Like "Include one more object to the scene", "customize the 3D configurator, switching the meshes"..

**Reply by elalish** [reply]
Also look at this [example](https://modelviewer.dev/examples/scenegraph/#hideShowMeshVariantsExample) for an approach that stays inside our public API.

**Reply by digityze2023** [reply]
This is a sort of "configuration" because you are switching between two meshes. Is it right? 
I can use it as sample for my devs.

**Reply by elalish** [reply]
Exactly - since we don't expose meshes directly, you apply separate materials to each thing or group of things you want to show/hide individually. Then mark them as BLEND and animate their alpha factor to show/hide.

**Reply by digityze2023** [reply]
Thanks a lot for your support!
